libbeat/common/kubernetes/metadata/pod.go

```diff
@@ -49,6 +49,8 @@ func (p *pod) Generate(obj kubernetes.Resource, opts ...FieldOptions) common.Map
 	}
 
 	out := p.resource.Generate("pod", obj, opts...)
+	// TODO: remove this call when moving to 8.0
```

Comment:
I thinks it's fair to create an issue for removing this TODO ;)

Reply:
https://github.com/elastic/beats/issues/16483

Reply:
+1, that said. I'm not quite convinced this is the way to go. Let's move this fix forward and discuss future steps in the opened issue